dtfs-central-api/src/v1/controllers/utilisation-report-service/post-add-fees-to-an-existing-payment-group.controller/index.test.ts

```diff
@@ -158,6 +158,36 @@ describe('post-fees-to-an-existing-payment-group.controller', () => {
       expect(res._getStatusCode()).toBe(HttpStatusCode.BadRequest);
     });
 
+    it("responds with a '400' if the fee records payment currency does not equal the payment group currency", async () => {
+      // Arrange
+      const paymentId = 3;
+      const aPaymentInUSD = PaymentEntityMockBuilder.forCurrency('USD').withId(paymentId).withFeeRecords([]).build();
+      const aFeeRecordWithAPaymentInUSD = FeeRecordEntityMockBuilder.forReport(utilisationReport)
+        .withId(2)
+        .withPaymentCurrency('GBP')
+        .withPayments([aPaymentInUSD])
+        .build();
```

Comment:
A few things:
- We need to set the status of this fee record in this test since there is validation in the endpoint of the fee record status to be in TO_DO status and we don't want this test to fail for that reason
- A fee record in TO_DO status won't have any payments attached, so please can we removed them and rename this to `aFeeRecordWithPaymentCurrencyGbp`
- The payment also needs a fee record to be allowed to have another fee record added so separately we need `aFeeRecordWithPaymentCurrencyUsd` which is attached to the payment and has a different id to the GBP one